### PR TITLE
Added LocalDate scalar

### PR DESCRIPTION
**Pull request by damikun**
Adding LocalDate scalar + Tests

Part of #2927

### CODE REVIEW

**Comment by PascalSenn** [comment]
Hi @damikun  

Thanks for the PR! 
I see you chose the Regex type as the base for the LocalDate scalar. I wonder if it would make more sense to have it resolved into a DateTime rather than a string. 
Just discussed something similar with @wonbyte  over in #3281 . What do you think?

**Comment by damikun** [comment]
Okay I can do that... I used string approach since RFC define only format and our source materials use also string... but I will rewrite to use DateTtime parser and use c# runtime.. Tests can be reused... I will ping you in day-two when i find time :P

**Comment by damikun** [comment]
Just for info... I need to add more tests + make it convertible between Offset and DT by default..

File: src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+using HotChocolate.Language;
+
+namespace HotChocolate.Types.Scalars
+{
+    /// <summary>
+    /// The `LocalDate` scalar type represents a ISO date string, represented as UTF-8
+    /// character sequences YYYY-MM-DD. The scalar follows the specification defined in
+    /// <a href="https://tools.ietf.org/html/rfc3339">RFC3339</a>
+    /// </summary>
+    public class LocalDateType  : ScalarType<DateTime, StringValueNode>
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs:14]:
```suggestion
    public class LocalDateType : ScalarType<DateTime, StringValueNode>
```

File: src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        private const string _localFormat = "yyyy-MM-dd";
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="LocalDateType"/> class.
+        /// </summary>
+        public LocalDateType()
+            : this(
+                WellKnownScalarTypes.LocalDate,
+                ScalarResources.LocalDateType_Description,
+                BindingBehavior.Implicit)
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs:25]: We need to register this type by default as explicit
```suggestion
                ScalarResources.LocalDateType_Description)
```

File: src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        }
+
+        public override IValueNode ParseResult(object? resultValue)
+        {
+            return resultValue switch
+            {
+                null => NullValueNode.Default,
+                string s => new StringValueNode(s),
+                DateTimeOffset d => ParseValue(d),
+                DateTime dt => ParseValue(new DateTimeOffset(dt)),
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs:48]: Just use ParseValue for DateTime too, see blow

File: src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        public override bool TrySerialize(object? runtimeValue, out object? resultValue)
+        {
+            switch (runtimeValue)
+            {
+                case null:
+                    resultValue = null;
+                    return true;
+                case DateTimeOffset dt:
+                    resultValue = Serialize(dt);
+                    return true;
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs:77]: Add a case for DateTime too

File: src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+                case string s when TryDeserializeFromString(s, out DateTime? d):
+                    runtimeValue = d;
+                    return true;
+                case DateTimeOffset:
+                    runtimeValue = resultValue;
+                    return true;
+                case DateTime dt:
+                    runtimeValue = new DateTimeOffset(
+                        dt.ToUniversalTime(),
+                        TimeSpan.Zero);
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs:100]: Just return datetime

File: src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+            switch (resultValue)
+            {
+                case null:
+                    runtimeValue = null;
+                    return true;
+                case string s when TryDeserializeFromString(s, out DateTime? d):
+                    runtimeValue = d;
+                    return true;
+                case DateTimeOffset:
+                    runtimeValue = resultValue;
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs:95]:
```suggestion
                case DateTimeOffset d:
                    runtimeValue = d.DateTime;
```

File: src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+                        dt.ToUniversalTime(),
+                        TimeSpan.Zero);
+                    return true;
+                default:
+                    runtimeValue = null;
+                    return false;
+            }
+        }
+
+        private static string Serialize(DateTimeOffset value)
```
Comment (PascalSenn) [inline src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs:108]: The trick to handle DateTime and DateTimeOffset is:
```suggestion
        private static string Serialize(IFormattable value)
```

**Review by PascalSenn — CHANGES_REQUESTED** [review]
Hi @damikun  This looks really good!!!!

File: src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs (hunk 7 above)
Comment (damikun) [reply src/HotChocolate/Core/src/Types.Scalars/LocalDateType.cs:107]: This is interesting i never heard abut IFormattable

**Review by PascalSenn — APPROVED** [review]

**Pull request merged**